src/figuresClasses.ts

```diff
@@ -2,10 +1,76 @@
+export interface Figure {
+  shape: string;
+  color: string;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  public shape = 'triangle';
+
+  constructor(
+    public color: string,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (
+      this.a <= 0 ||
+      this.b <= 0 ||
+      this.c <= 0 ||
+      this.a >= this.b + this.c ||
+      this.b >= this.a + this.c ||
+      this.c >= this.b + this.a
+    ) {
+      throw new Error(`Sides ${a}, ${b} and ${c} can not form a triangle`);
+    }
+  }
+
+  getArea(): number {
+    const s = (1 / 2) * (this.a + this.b + this.c);
+
+    return (
+      Math.round(
+        Math.sqrt(s * (s - this.a) * (s - this.b) * (s - this.c)) * 100,
+      ) / 100
+    );
+  }
+}
 
-export class Triangle implements Figure {}
+export class Circle implements Figure {
+  public shape = 'circle';
 
-export class Circle implements Figure {}
+  constructor(
+    public color: string,
+    public radius: number,
+  ) {
+    if (radius <= 0) {
```

Comment:
Use `this.radius` instead of `radius` to correctly reference the instance property in the error check.